rb/lib/selenium/webdriver/common/service.rb

```diff
@@ -87,16 +88,25 @@ def initialize(path: nil, port: nil, log: nil, args: nil)
       end
 
       def launch
-        @executable_path ||= begin
-          default_options = WebDriver.const_get("#{self.class.name&.split('::')&.[](2)}::Options").new
-          DriverFinder.new(default_options, self).driver_path
-        end
+        @executable_path ||= env_path || find_driver_path
         ServiceManager.new(self).tap(&:start)
       end
 
       def shutdown_supported
         self.class::SHUTDOWN_SUPPORTED
       end
+
+      def find_driver_path
+        default_options = WebDriver.const_get("#{self.class.name&.split('::')&.[](2)}::Options").new
+        DriverFinder.new(default_options, self).driver_path
+      end
+
+      def env_path
+        class_name = self.class.name&.split('::')&.[](2)&.downcase
+        driver_name = class_name == 'firefox' ? 'gecko' : class_name
+        parsed_driver = "SE_#{driver_name&.upcase}DRIVER"
```

Comment:
I think edge is referred to as msedgedriver. So it might need more guards. But I'm not 100% on this.

Reply:
Let's see now with the new env variables because last run the ie driver test did not fail, so I want to double check